python/lsst/daf/butler/_butlerRepoIndex.py

```diff
@@ -136,6 +140,23 @@ def get_known_repos(cls) -> Set[str]:
             return set()
         return set(repo_index)
 
+    @classmethod
+    def get_failure_reason(cls) -> str:
+        """Return possible reason for failure to return repository index.
+
+        Returns
+        -------
+        reason : `str`
+            If there is a problem reading the repository index, this will
+            contain a string with an explanation. Empty string if everything
+            worked.
+        """
+        try:
+            cls._read_repository_index_from_environment()
```

Comment:
Hmm, I guess this assumes that it is supposed to fail in exactly the same way every time? With transient filesystem issues it may not be true, would it be better to save the original exception message and return it here?

Reply:
Hmm. There is a cache for the contents of the URI so I probably could cache the contents of the exception in a different cache in the class. That should be okay for a transient failure assuming people don't call this out of sequence.

Reply:
Okay. I've changed it to a simple cache that resets on each call and that seems to work okay.